soroban-env-host/src/host.rs

```diff
@@ -1738,12 +1832,29 @@ impl VmCallerEnv for Host {
         &self,
         _vmcaller: &mut VmCaller<Host>,
         k: RawVal,
+        t: StorageType,
     ) -> Result<Void, HostError> {
-        let key = self.contract_data_key_from_rawval(k)?;
+        let key = self.contract_data_key_from_rawval(k, t.try_into()?)?;
         self.0.storage.borrow_mut().del(&key, self.as_budget())?;
         Ok(RawVal::VOID)
     }
 
+    // Notes on metering: covered by components
+    fn bump_contract_data(
+        &self,
+        _vmcaller: &mut VmCaller<Host>,
+        k: RawVal,
+        t: StorageType,
+        min: U32Val,
+    ) -> Result<Void, HostError> {
+        return Err(self.err(
```

Comment:
Maybe file an issue on this mentioning that we need to figure out the exact fee semantics and delta-transfer mechanism?

Reply:
https://github.com/stellar/rs-soroban-env/issues/841